### ISSUE TITLE
Support indented code blocks

### ISSUE THREAD

**Issue by smikitky** [smikitky/chatgpt-md-translator#7]
This tool replaces code blocks with placeholders before calling the API. But currently it does not understand indented code blocks within a list.

    1. Install foobar.

       ````bash
       $ npm install foobar
       ````

    2. Run it.

React docs don't have code blocks like this, but some docs do use this pattern aggressively, so we must deal with this.

**Comment by smikitky** [comment]
@DanRoscigno Perhaps you can make a draft PR so I can take a look. If you can, please use the `v1.0` branch as the base.

**Comment by smikitky** [comment]
I implemented this in bcb9e242a5b7bd011abd3ac1ed24c257086d24fd. For now, you can try this like so:

```
git checkout v1.0
npm run build
node lib/index -f1000 -m3 file-to-translate.md
```

I intend to release 1.0 once [this problem](https://github.com/smikitky/markdown-gpt-translator/pull/5#issuecomment-1650876402) has been resolved, but even if @mengxin doesn't notice this, I will rename this package and release 1.0 within a few weeks.

**Comment by DanRoscigno** [comment]
Thanks so much @smikitky , I ran through the install process for release 1.0 and tested translating a file, the codeblocks are now generated perfectly and all of the content is translated.  I will find someone who reads Japanese to verify the translation and start working on a glossary.
